api/src/DataPersister/Util/PropertyChangeListener.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace App\DataPersister\Util;
+
+use Closure;
+use ReflectionFunction;
+
+class PropertyChangeListener {
+    private function __construct(
+        private Closure $extractProperty,
+        private Closure $afterAction
+    ) {
+    }
+
+    /**
+     * @throws \ReflectionException
+     */
+    public static function of(
+        Closure $extractProperty,
+        ?Closure $afterAction = null
+    ): PropertyChangeListener {
```

Comment:
Auch diesen Listener hätte ich jetzt intuitiv eher mit abstrakten Methoden `extractProperty` und `afterChange` (in Zukunft vielleicht sogar mal `beforeChange`) umgesetzt statt mit Closures. Hast du dich bewusst für Closures entschieden?